### PR TITLE
networking.greTunnels: support ip6gre*

### PR DESCRIPTION
**Pull request by jpathy**
###### Description of changes

Added ip6gre support.

### CODE REVIEW

File: nixos/modules/tasks/network-interfaces.nix [hunk 1]
```diff
@@ -1058,10 +1064,12 @@ in
...
 
           type = mkOption {
-            type = with types; enum [ "tun" "tap" ];
+            type = with types; enum [ "tun" "tap" "tun6" "tap6" ];
             default = "tap";
             example = "tap";
-            apply = v: if v == "tun" then "gre" else "gretap";
+            apply = v: if stringLength v == 3
+              then (if v == "tun" then "gre" else "gretap")
+              else (if v == "tun6" then "ip6gre" else "ip6gretap");
```
Comment (pennae) [inline nixos/modules/tasks/network-interfaces.nix:1072]:
```suggestion
            apply = v: {
              tun = "gre";
              tap = "gretap";
              tun6 = "ip6gre";
              tap6 = "ip6gretap";
            }.${v};
```
this is a little more verbose, but also a little more robust should more flags be added to the enum in the future.

**Review by pennae — COMMENTED** [review]
could you also add a test to the `networking.nix` set for this? would be good to have assurance that it doesn't break by accident in the future.

File: nixos/modules/tasks/network-interfaces.nix (hunk 1 above)
Comment (jpathy) [reply nixos/modules/tasks/network-interfaces.nix]: I have locally tested the new test and applied your suggestion.

**Comment by pennae** [comment]
@ofborg test networking

**Pull request merged**